File: src/string/ascend_descend.py
def ascend_descend(length: int, min: int, max: int):
  if length <= 0 or min > max: return ""
  if min == max: 
    return str(min) * length if min > 0 else (str(min) * length)[:length]
  num = min
  num_str = str(num)
  string = ""
  is_going_up = min < max
  index = 0
  while len(string) < length:
    if is_going_up:
      num += 1
      if num == max: is_going_up = False
    else:
      num += -1
      if num == min: is_going_up = True
    string += num_str
    num_str = str(num)
    index += 1
  return string

File: src/string/ascend_descend.ts
export default function ascendDescend(length: number, min: number, max: number) {
  if (length <= 0 || min > max) return "";
  let string = "";
  let number = min;
  let char = number.toString();
  if (min === max) return char.repeat(length);
  let isGoingUp = number < max;
  for (let index = 0; string.length < length; index++) {
    if (isGoingUp) {
      number++;
      if (number == max) isGoingUp = false;
    } else {
      number--;
      if (number == min) isGoingUp = true;
    }
    string += char;
    char = number.toString();
  }
  return string.substring(0, length);
}
